src/main/webapp/js/script.js

```diff
@@ -226,22 +249,23 @@ function highlightAffectedCourses(affectedCourses){
 function generateSequenceObject(callback){
 	var semesterList = [];
 	var count = 0;
-	var onFinish = function(semesterObject){
-        if(semesterObject){
-            semesterList.push(semesterObject);
+	var numberOfTerms = $(".sequenceContainer .term").length;
+	var onFinish = function(sequenceObject){
```

Comment:
Change this back to semesterObject. Sry if I misled you here, but the object in question represents a semester, not a whole sequence